Updates the orb testing/publishing workflow

rotated8 opened the pull request:
- Updates the orb-tools version
- Automatically push dev versions of the orb if they pass lint and pack
- Updates the README to explain the automatic behavior
- Adds a prototype job section, for future testing

rotated8 committed:
Updates the orb testing/publishing workflow

- Updates the orb-tools version
- Automatically push dev versions of the orb if they pass lint and pack
- With approval, creates prod versions for tagged releases
- Updates the README to explain the automatic behavior
- Adds a prototype integration testing section

jrgriffiniii commented on README.md:
```diff
@@ -3,7 +3,15 @@
+that was built.
+Orb versions that begin with `dev:` can be overwritten by anyone, and only exist for 90 days.
+
+Additionally, publishing dev and production versions of the orb can be done manually:
```
👍

jrgriffiniii commented:
This is superb, thank you very much for introducing the new workflow! Please forgive me, but I would like to first take the time to test this on a local repository in order to understand this properly before merging.

rotated8 commented:
This PR has been updated to depend on PR #41.

rotated8 committed:
Updates to latest orb-tools version

Main difference for us is orb-tools no longer has executors, and the shellcheck command is gone.
